beacon/engine/types.go

```diff
@@ -290,6 +294,7 @@ func ExecutableDataToBlockNoHash(data ExecutableData, versionedHashes []common.H
 		BlobGasUsed:      data.BlobGasUsed,
 		ParentBeaconRoot: beaconRoot,
 		RequestsHash:     requestsHash,
+		SlotNumber:       data.SlotNumber,
```

Comment:
This will set correct slot number, but for `newPayload` call, not for the `forkChoiceUpdated`, and `forkChoiceUpdated` is _the_ call to initiate block building process.